homeassistant/components/jellyfin/media_player.py

```diff
@@ -0,0 +1,489 @@
+                    | MediaPlayerEntityFeature.PLAY
+                    | MediaPlayerEntityFeature.STOP
+                )
+        except KeyError:
+            pass
+
+        return ret
+
+    @property
+    def state(self) -> str:
+        """State of the player."""
+        if not self._data:
+            return STATE_OFF
+
+        if not ("NowPlayingItem" in self._data):
+            return STATE_IDLE
+
+        if self._data["PlayState"]["IsPaused"]:
+            return STATE_PAUSED
+
+        return STATE_PLAYING
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def is_volume_muted(self) -> bool:
+        """Boolean if volume is currently muted."""
+        return bool(self._data["PlayState"]["IsMuted"])
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def media_episode(self) -> str | None:
+        """Episode of current playing media, TV show only."""
+        return str(self._data["NowPlayingItem"]["IndexNumber"])
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def media_season(self) -> str | None:
+        """Season of current playing media, TV show only."""
+        return str(self._data["NowPlayingItem"]["ParentIndexNumber"])
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def media_series_title(self) -> str | None:
+        """Title of series of current playing media, TV show only."""
+        return str(self._data["NowPlayingItem"]["SeriesName"])
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def media_title(self) -> str | None:
+        """Title of current playing media."""
+        return str(self._data["NowPlayingItem"]["Name"])
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def media_position(self) -> int | None:
+        """Position of current playing media in seconds."""
+        return int(self._data["PlayState"]["PositionTicks"] / 10000000)
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def media_duration(self) -> int | None:
+        """Duration of current playing media in seconds."""
+        return int(self._data["NowPlayingItem"]["RunTimeTicks"] / 10000000)
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def media_content_id(self) -> str | None:
+        """Content ID of current playing media."""
+        return str(self._data["NowPlayingItem"]["Id"])
+
+    @property  # type: ignore[misc]
+    @needs_data
+    def volume_level(self) -> float | None:
+        """Volume level of the media player (0..1)."""
+        return float(self._data["PlayState"]["VolumeLevel"] / 100)
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return bool(self._data)
+
+    def media_seek(self, position: int) -> None:
+        """Send seek command."""
+        self._client.jellyfin.sessions(
+            handler=f"/{self._id}/Playing/Seek",
+            action="POST",
+            params={"seekPositionTicks": int(position * 10000000)},
+        )
```

Comment:
For these kind of commands you are likely going to get the feedback from the maintainers that this should be handled in the API and not in the Home Assistant integration. I got the same kind of feedback.

You can submit PRs to https://github.com/jellyfin/jellyfin-apiclient-python and they usually respond very quickly.

Reply:
Will do.

Then I can also improve the artwork API. That could allow the integration to provide the watched checkmarks and percent.